### ISSUE TITLE
Request: pass less data into ReadFrom.Configuration

### ISSUE THREAD

**Issue by lonix1** [serilog/serilog-settings-configuration#117]
This is the typical setup:

`````csharp
var configuration = getConfiguration();
var logger = new LoggerConfiguration()
  .ReadFrom.Configuration(configuration);
  .CreateLogger();
`````

However that passes the entire `appsettings.json` into `ReadFrom.Configuration()`. Which is not a good idea - Serilog should get only that config data it needs, no more.

It would be more robust to OPTIONALLY replace this:

`````csharp
.ReadFrom.Configuration(configuration);
`````
with this:

`````csharp
.ReadFrom.Configuration(configuration.GetSection("Serilog"));
`````

Both forms would be allowed to ensure backwards compatibility. Serilog would then need to decide whether the json tree it gets has "Serilog" as the root node, or not, and proceed accordingly.

**Comment by tsimbalar** [comment]
Hi and thank you for opening an issue.

I believe both overloads already exist :

```csharp
public static LoggerConfiguration Configuration(
    this LoggerSettingsConfiguration settingConfiguration,
    IConfiguration configuration,
    DependencyContext dependencyContext = null)
    {}

public static LoggerConfiguration ConfigurationSection(
    this LoggerSettingsConfiguration settingConfiguration,
    IConfigurationSection configSection,
    DependencyContext dependencyContext = null)
    {}
```

Doesn't that cover your remarks ?

Thanks

**Comment by MV10** [comment]
@lonix1 by way of explanation, the main use-case for passing the entire configuration is to allow various Serilog libraries to access shared configuration information. The most obvious example is a database connection string section.

**Comment by lonix1** [comment]
@MV10 Yes that makes sense I suppose. Though a better solution would be to pass such data within the "Serilog" config block. An issue of convenience vs robustness/security.

**Comment by lonix1** [comment]
Also true. And in that case there would be duplication of such connection strings which is a recipe for trouble.

I'm glad the overload exists because it's good enough for my use case.
